### DISCUSSION TITLE
Application Process for the GSoC 2023

### DISCUSSION THREAD

**Discussion by xansh** [alan-turing-institute/sktime#4046] [General]
Hi everyone

I am a final year Computer Science undergraduate student at the Kalinga Institute of Industrial Technology, India.

I have a solid foundational background in Computer Science and mathematics and have done several projects related to application development, machine learning and computer vision in the past. I’m familiar with MERN stack (using Node and SQL/NoSQL DBs), Tensorflow, Pytorch.

I am interested in contributing to sktime. If the organization is participating in Google Summer of Code this year, I’m interested in applying for the program too.

As I’m completely new to sktime, I’d be grateful if you can guide me as to how I can get more involved (things I should read before diving into the code and issues which are suitable for a beginner like me)? I’m willing to learn any new technologies and pick up challenging projects.

Looking forward to a response.

Thanks!

Ansh Kumar
Final year UG student, Computer Science
Kalinga Institute of Industrial Technology

**Comment by fkiraly** [comment]
Welcome to `sktime`, Ansh!

Here's a summary of how to get started with contributing:
https://www.sktime.net/en/stable/get_started.html

For GSoC, you are quite early, the organisations have not applied yet nor have they been confirmed.
However, we are definitely planning to apply (as an organisation), and if successful, the process will likely be exactly as 2022, with different dates and new project ideas:
https://github.com/sktime/mentoring/blob/main/internships/gsoc2022.md

**Reply by nikitashrivastava29** [reply]
Hi,

I went through both the links which you've provided for the previous question. I was hoping if there were any repos that I could maybe get my hands dirty with in the mean time (while the organization registration happens).

P.S: Found this link (https://github.com/sktime/sktime/issues/1147). Is this a good starting point?

Thanks,
Nikita

**Reply by fkiraly** [reply]
yes, #1147 has a collection of starter issues, although it might not be most up-to-date.
There should be a selection of good first issues in this repo though.

**Comment by fkiraly** [comment]
Project ideas for 2023 - tentative
https://github.com/sktime/mentoring/blob/main/internships/projects_2023.md

**Reply by nikitashrivastava29** [reply]
Thanks @fkiraly!

**Reply by nikitashrivastava29** [reply]
Hi @fkiraly,

I was interested in discussing the project idea of "Probabilistic Forecasting and Supervised Learning" with you as I saw that you were one of the mentors. Where should I contact you regarding any follow-up ideas and discussions? Is Github a proper platform or should I mail you?

Regards,
Nikita

**Reply by nikitashrivastava29** [reply]
Also, would sktime be participating in Outreachy this year?

**Reply by fkiraly** [reply]
> Also, would sktime be participating in Outreachy this year?

No, sktime is not participating in Outreachy this year.

> Where should I contact you regarding any follow-up ideas and discussions? Is Github a proper platform or should I mail you?

The recommendation is to get started with contributing to `sktime` and getting to know the community (not just one person). The link for "getting started" is above; there are also weekly community collaboration sessions on Fridays (4pm, discord) whre you can meet other contributors, users, and developers.

**Reply by nikitashrivastava29** [reply]
Thanks a lot @fkiraly, I'll be sure to join in on next Friday's meeting and contribute more to issues!

**Comment by fkiraly** [comment]
IMPORTANT: sktime is not participating in GSoC 2023 this year.
This should be clear from it not appearing on the orgs page, or being selectable in the application documents (side note, it appears ML frameworks like scikit-learn weren't among the accepted orgs this year more generally).

Kindly be aware of this to avoid preparing an application that you cannot submit.

There will be paid internships (tba soon) and there is the unpaid mentoring programme: https://github.com/sktime/mentoring
